ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -189,13 +209,17 @@ public void setUnderlineColor(ReactEditText view, @Nullable Integer underlineCol
     }
   }
 
-  @ReactProp(name = "textAlign")
-  public void setTextAlign(ReactEditText view, int gravity) {
+  @ReactProp(name = ViewProps.TEXT_ALIGN)
+  public void setTextAlign(ReactEditText view, @Nullable String textAlign) {
+    Map<String, Integer> constants = TEXT_ALIGN_CONSTANTS.get(ViewProps.TEXT_ALIGN);
+    int gravity = constants.get(textAlign);
     view.setGravityHorizontal(gravity);
   }
 
   @ReactProp(name = "textAlignVertical")
-  public void setTextAlignVertical(ReactEditText view, int gravity) {
+  public void setTextAlignVertical(ReactEditText view, @Nullable String textAlignVertical) {
+    Map<String, Integer> constants = TEXT_ALIGN_CONSTANTS.get("textAlignVertical");
```

Comment:
I wouldn't use hash maps for this in java. They tend not to be very efficient (compering to alternative approaches) when you have a small number of keys. I'd recommend to at least have a separate maps for `TEXT_ALIGN_CONSTANTS` and `TEXT_ALIGN_VERTICAL_CONSTANTS` or even resign from the secondary level map here and just having a bunch of if blocks similarily to how it's handled here: [ReactTextViewManager.java](https://github.com/facebook/react-native/blob/a0268a7bfc8000b5297d2b50f81e000d1f479c76/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java#L60)